### PR TITLE
db:seed and db:unseed

### PR DESCRIPTION
**Pull request by ataube**
This is a working PR based on #114. Merged with master and fixed the tests. 

Many thanks to @Americas for the initial work. Lets get it shipped...

### CODE REVIEW

**Comment by ataube** [comment]
Damn, there is an issue with mysql. I guess it's the countTable() function, using proper parameters should fix that...

**Comment by sdepold** [comment]
:cool: are you going to fix it?

File: test/support/assets/seeders/20111117063700-seedPerson.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+var nodeify = require('nodeify');
+
+module.exports = {
+  up: function (migration, DataTypes, done) {
+    nodeify(migration.bulkInsert('Person', [{
```
Comment (janmeier) [inline test/support/assets/seeders/20111117063700-seedPerson.js]: No need to `nodify`, you should be able to just return the promise

**Comment by ataube** [comment]
Cool thanks for the feedback, I will fix it. Any suggestions for the countTable() function. Is it the right approach to fix the mysql error if I use proper query replacements? 
http://docs.sequelizejs.com/en/latest/docs/raw-queries/

**Comment by janmeier** [comment]
pass `{ type: sequelize.QueryTypes.SELECT}`, then you will get the result directly, instead of an array

**Comment by ataube** [comment]
ok guys, this PR is ready to be merged. There is still one test failing, same as master ;-)

@janmeier I still use nodify just about consistency, all other seeders/migrations use the same pattern. If this is still an issue for you, I would suggest opening a new issue for that.

**Comment by sdepold** [comment]
I restarted one of the builds

File: lib/helpers/umzug-helper.js [hunk 2]
```diff
@@ -4,32 +4,43 @@ var path    = require('path');
...
+  getStorage: function (type) {
+    return this.getStorageOption(type + 'Storage', 'sequelize');
   },
 
-  getStoragePath: function () {
-    var fallbackPath = path.join(process.cwd(), 'sequelize-meta.json');
+  getStoragePath: function (type) {
+    var fallbackPath = path.join(process.cwd(), storageJsonName[type]);
+
+    console.log(fallbackPath);
```
Comment (sdepold) [inline lib/helpers/umzug-helper.js]: This is a leftover I guess?

File: test/support/assets/seeders/20111117063700-seedPerson.js (hunk 1 above)
Comment (sdepold) [inline test/support/assets/seeders/20111117063700-seedPerson.js]: IIRC it contains `nodify` to support the old versions of sequelize (the CLI basically supports all version of sequelize).

**Comment by sdepold** [comment]
@Americas @ataube :ok_hand: awesome stuff!

File: lib/assets/seeders/skeleton.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+    /*
+      Add altering commands here.
+      Return a promise to correctly handle asynchronicity.
+
+      Example:
+      return queryInterface.createTable('users', { id: Sequelize.INTEGER });
```
Comment (sdepold) [inline lib/assets/seeders/skeleton.js]: Should this do something with bulkInsert?

File: lib/assets/seeders/seed-table.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+'use strict';
```
Comment (sdepold) [inline lib/assets/seeders/seed-table.js]: Where are we using this file?

**Comment by sdepold** [comment]
functionality wise it seems to work nicely :+1:

File: lib/helpers/umzug-helper.js (hunk 2 above)
Comment (ataube) [inline lib/helpers/umzug-helper.js]: good spot, this one shouldn't be there...

File: lib/assets/seeders/skeleton.js (hunk 3 above)
Comment (ataube) [inline lib/assets/seeders/skeleton.js]: correct, the example should be seeder specific...

File: lib/assets/seeders/seed-table.js (hunk 4 above)
Comment (ataube) [inline lib/assets/seeders/seed-table.js]: This seems like a copy past mistake, I couldn't find any reference to this file => deleting it

**Comment by sdepold** [comment]
:cool: I'm going to squash the commits, fix a tiny typo and merge it manually

**Comment by sdepold** [comment]
Just released as 1.9.0. Thanks a lot for your work!

**Comment by durazzocl** [comment]
I've just test it and it seems you should update your template.

QueryInterface.prototype.bulkInsert = function(tableName, records, options, attributes) {
  options.type = QueryTypes.INSERT;
  var sql = this.QueryGenerator.bulkInsertQuery(tableName, records, options, attributes);
  return this.sequelize.query(sql, options);
};

This function implies that "options" is not null. But your template indicates that options is "optionnal"

Example:
      return queryInterface.bulkInsert('Person', [{
        name: 'John Doe',
        isBetaMember: false
      }]);

So without it, it throw an exception.

Except that, it works perfectly. Thank you very much. It's a very usefull feature

**Comment by sdepold** [comment]
Meh, good point. It's a pity that options is not optional -.-

**Pull request closed without merging**